src/java/org/apache/cassandra/cql3/selection/ListSelector.java

```diff
@@ -90,6 +112,17 @@ public void reset()
             elements.get(i).reset();
     }
 
+    @Override
+    public boolean isTerminal()
+    {
+        for (int i = 0, m = elements.size(); i < m; i++)
+        {
+            if(!elements.get(i).isTerminal())
```

Comment:
Nit: formatting. That missed whitespace is also in the `isTerminal` implementations in `SetSelector`, `TupleSelector` and `UserTypeSelector`.
```suggestion
            if (!elements.get(i).isTerminal())
```
```suggestion
            if(!elements.get(i).isTerminal())
```